src/App.tsx

```diff
@@ -1,14 +1,54 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
+import { useDispatch } from 'react-redux';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { useAppSelector } from './app/hooks';
+import { getTodos } from './api';
+import { actions as todosActions } from './features/todos';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { RootState } from './app/store';
+
+const filteredTodosSelector = (state: RootState) => {
+  const { todos } = state.todos;
+  const { query, status } = state.filter;
+
+  return todos.filter(todo => {
+    const includesQuery = todo.title.toLowerCase().includes(query.toLowerCase());
+
+    switch (status) {
+      case 'all':
+        return includesQuery;
+
+      case 'active':
+        return todo.completed === false && includesQuery;
+
+      case 'completed':
+        return todo.completed && includesQuery;
+
+      default:
+        return true;
+    }
+  });
+};
 
 export const App: React.FC = () => {
+  const dispatch = useDispatch();
+  const selectedTodo = useAppSelector(state => state.currentTodo);
```

Comment:
Create selectors and call them here, like you do in 41 line
```suggestion
  const selectedTodo = useAppSelector(getSelectedTodo);
```